Code/GraphMol/Deprotect/Deprotect.cpp

```diff
@@ -0,0 +1,107 @@
+//
+//  Copyright (C) 2020 Brian P Kelley, Joann Prescott-Roy
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "Deprotect.h"
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/ChemReactions/ReactionParser.h>
+
+#include <boost/smart_ptr.hpp>
+
+namespace RDKit {
+
+DeprotectData::DeprotectData(const std::string &deprotection_class,
+			     const std::string &reaction_smarts,
+			     const std::string &abbreviation,
+			     const std::string &full_name):
+
+  deprotection_class(deprotection_class), 
+  reaction_smarts(reaction_smarts), 
+  abbreviation(abbreviation),
+  full_name(full_name), 
+  rxn(RxnSmartsToChemicalReaction(reaction_smarts))
+{
+  if (rxn.get())
+    rxn->initReactantMatchers();
+}
+
+const std::vector<DeprotectData> &getDeprotections() {
+  static const std::vector<DeprotectData> deprotections {
```

Comment:
And this initialization is even thread safe in modern C++. nice.

Reply:
Yes, yes it is!